pkg/setup/vmxenabled.go

```diff
@@ -442,6 +442,9 @@ func downloadFile(client *http.Client, url, destDir string) error {
 	if err != nil {
 		return err
 	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("failed to download %s, status code: %d", url, resp.StatusCode)
+	}
 	defer resp.Body.Close()
```

Comment:
We should put "defer Close()" just after we confirm non-nil `err`.

```suggestion
	defer resp.Body.Close()
	if resp.StatusCode != http.StatusOK {
		return fmt.Errorf("failed to download %s, status code: %d", url, resp.StatusCode)
	}
```

Reply:
It would be helpful to show the contents of `resp.Body` in order to investigate the cause of the error.
Please see the example of https://pkg.go.dev/net/http#Get .